library/core/src/fmt/mod.rs

```diff
@@ -277,55 +279,117 @@ impl<'a> Formatter<'a> {
 #[stable(feature = "rust1", since = "1.0.0")]
 #[derive(Copy, Clone)]
 pub struct Arguments<'a> {
-    // Format string pieces to print.
-    pieces: &'a [&'static str],
+    /// The number of string pieces and place holders combined.
+    ///
+    /// For example:
+    ///  - 1 for `format_args!("abc")`
+    ///  - 2 for `format_args!("abc{}")`
+    ///  - 3 for `format_args!("abc{}xyz")`
+    ///  - 4 for `format_args!("abc{}xyz{}")`
+    ///  - 5 for `format_args!("abc{}xyz{}123")`
+    ///
+    ///  The number of placeholders is `num_parts / 2`.
+    ///  The number of string pieces is `(num_parts + 1) / 2`.
```

Comment:
For me the case of `format_args!("{}")` is missing here, mentioning that an empty string is prepended in that situation. (Source: https://github.com/rust-lang/rust/blob/master/compiler/rustc_ast_lowering/src/format.rs#L373-L380)

Otherwise the provided formulas dont make sense